PSReadLine/PlatformWindows.cs

```diff
@@ -614,4 +611,350 @@ public override void BlankRestOfLine()
     [DllImport("user32.dll", SetLastError = true)]
     [return: MarshalAs(UnmanagedType.Bool)]
     internal static extern bool SystemParametersInfo(uint uiAction, uint uiParam, ref bool pvParam, uint fWinIni);
+
+    [DllImport("kernel32.dll", SetLastError = true, EntryPoint = "GetConsoleProcessList")]
+    private static extern uint native_GetConsoleProcessList([In, Out] uint[] lpdwProcessList, uint dwProcessCount);
+
+    private static uint[] GetConsoleProcessList()
+    {
+        int size = 100;
+        uint[] pids = new uint[size];
+        uint numPids = native_GetConsoleProcessList(pids, (uint) size);
+
+        if (numPids > size)
+        {
+            size = (int) numPids + 10; // a bit extra, since we may be racing attaches.
+            pids = new uint[size];
+            numPids = native_GetConsoleProcessList(pids, (uint) size);
+        }
+
+        if ((0 == numPids) || (numPids > size))
+        {
+            return null; // no TerminateOrphanedConsoleApps for you, sorry
+        }
+
+        Array.Resize(ref pids, (int) numPids);
+        return pids;
+    }
+
+    // If the TerminateOrphanedConsoleApps option is enabled, this is the list of PIDs
+    // that are allowed to stay attached to the console (effectively the current process
+    // plus ancestors).
+    private static uint[] _allowedPids;
+
+    internal static void SetTerminateOrphanedConsoleApps(bool enabled)
+    {
+        if (enabled)
+        {
+            _allowedPids = GetConsoleProcessList();
+        }
+        else
+        {
+            _allowedPids = null;
+        }
+    }
+
+    private static bool ItLooksLikeWeAreInTerminal()
+    {
+        return !String.IsNullOrEmpty(Environment.GetEnvironmentVariable("WT_SESSION"));
+    }
+
+    [DllImport("kernel32.dll", SetLastError = true)]
+    private static extern IntPtr GetConsoleWindow();
+
+    internal enum TaskbarStates
+    {
+        NoProgress    = 0,
+        Indeterminate = 0x1,
+        Normal        = 0x2,
+        Error         = 0x4,
+        Paused        = 0x8,
+    }
+
+    internal static class TaskbarProgress
+    {
+        [ComImport()]
+        [Guid("ea1afb91-9e28-4b86-90e9-9e9f8a5eefaf")]
+        [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+        private interface ITaskbarList3
+        {
+            // ITaskbarList
+            [PreserveSig]
+            int HrInit();
+
+            [PreserveSig]
+            int AddTab( IntPtr hwnd );
```

Comment:
Minor style thing. Can you please change the following members too?
```suggestion
            int AddTab(IntPtr hwnd);
```